### PR TITLE
Add documentation for ListItem

### PR DESCRIPTION
**Pull request by joannaquu**
### Platforms Impacted
- [ ] iOS
- [ ] visionOS
- [ ] macOS

### Description of changes

Adding documentation for `ListItem`

### Verification

See https://github.com/joannaquu/fluentui-apple/blob/joannaqu/list-item-doc/ios/docs/Controls/ListItem.md for rendered markdown

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/fluentui-apple/pull/2017)

### CODE REVIEW

**Review by alexanderboswell — APPROVED** [review]

**Comment by alexanderboswell** [comment]
should we add any details on the list accessory type?

File: ios/docs/Controls/ListItem.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# ListItem
+
+## Overview
+`ListItem` can be used either in a `List` or standalone to represent a single item within a vertical container. It can display a title, optional subtitle, optional footer, optional leading content view, and optional trailing content view. You can also customize the top and bottom separators, accessory type, and action. When an `action` parameter is passed in, `ListItem` will render as a `Button`. When `action` is nil or no `action` parameter is passed in, `ListItem` will render as a `View`.
```
Comment (anandrajeswaran) [inline ios/docs/Controls/ListItem.md:4]: Should `List` link to SwiftUI List documentation to make clear that there is no Fluent List control (at least yet)

File: ios/docs/Controls/ListItem.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# ListItem
+
+## Overview
```
Comment (anandrajeswaran) [inline ios/docs/Controls/ListItem.md:3]: Should we have a link somewhere to https://developer.microsoft.com/en-us/fluentui#/controls/ios/listcells? Would guess for the most part people using this documentation would be working from Figmas/designs using terminology like that?

**Review by anandrajeswaran — APPROVED** [review]

File: ios/docs/Controls/ListItem.md (hunk 2 above)
Comment (joannaquu) [reply ios/docs/Controls/ListItem.md:3]: that's still fluent1 and it also mentions `TableViewCell`, do we still want to maintain that `ListItem` is `TableViewCell`'s counterpart in swiftui?

**Review by mischreiber — APPROVED** [review]

File: ios/docs/Controls/ListItem.md (hunk 2 above)
Comment (joannaquu) [reply ios/docs/Controls/ListItem.md:3]: will follow up

**Pull request merged**